### PR TITLE
Example handling form data

### PR DESCRIPTION
**Pull request by AlexanderElias**
<h2>Complete example handling form data</h2>

In regards to ticket number <a href="https://github.com/nickel-org/nickel.rs/issues/240#issuecomment-127655614">#240</a>

If you would like me to strip the cargo package before upload and just leave the "views" and "main.rs" just let me know. 

All that needs to be added now is the url parse.

### CODE REVIEW

**Comment by Ryman** [comment]
If you have the nested cargo package, does `cargo test` in the base `nickel` directory still trigger the test? If not then I think you should just flatten the directories.

Can you update the commit message to adhere to [this convention](https://github.com/ajoslin/conventional-changelog/blob/master/conventions/angular.md)?

File: examples/contact_form/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#[macro_use] extern crate nickel;
+use nickel::{Nickel, HttpRouter};
+use std::collections::HashMap;
+
+// Required For Post
+extern crate hyper;
```
Comment (Ryman) [inline examples/contact_form/src/main.rs]: Is this import required?

File: examples/contact_form/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+use std::io::Read;
+// Required For Post
+
+fn main() {
+
+    //Implement A Server
+    let mut server = Nickel::new();
```
Comment (Ryman) [inline examples/contact_form/src/main.rs]: nit: weird spacing after this line, also prefer not to have blank lines after opening braces (after main, and within each `middleware!`)

File: examples/contact_form/src/main.rs (hunk 1 above)
Comment (AlexanderElias) [inline examples/contact_form/src/main.rs]: I had received a weird issue without it the first time. But I cleaned and then recompiled and now no issue. I will remove it.

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were style issues with your Pull Request:

 * Commit: 3649d7e35becaaa05863b4e7b7473a05db32217b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: e21075b1f15387a9f1d85b5ac37bc7af4771dfe4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: c41c70d98a33813b0bcd257ffeda379ea2a3c17a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/nickel-org/nickel.rs/blob/master/contributing.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were style issues with your Pull Request:

 * Commit: 3649d7e35becaaa05863b4e7b7473a05db32217b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: e21075b1f15387a9f1d85b5ac37bc7af4771dfe4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: c41c70d98a33813b0bcd257ffeda379ea2a3c17a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

 * Commit: 6822dc344683510d7f4def3ec78b8ff659026848
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: adc2191dd629adf6cbd2db54d5a3435c7e6a8d75
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/nickel-org/nickel.rs/blob/master/contributing.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were style issues with your Pull Request:

 * Commit: 3649d7e35becaaa05863b4e7b7473a05db32217b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: e21075b1f15387a9f1d85b5ac37bc7af4771dfe4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: c41c70d98a33813b0bcd257ffeda379ea2a3c17a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

 * Commit: 6822dc344683510d7f4def3ec78b8ff659026848
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: adc2191dd629adf6cbd2db54d5a3435c7e6a8d75
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 72 characters

 * Commit: 5804365af0ebbcb27f54993186a56ecef958f089
  * Your subject line is longer than 72 characters

Guidelines are available at https://github.com/nickel-org/nickel.rs/blob/master/contributing.md

---
This message was auto-generated by https://gitcop.com

**Comment by AlexanderElias** [comment]
Sorry about the mess I made I am newish to git. I think I finally got the convention down. What do we do about merge commits though? Or was I doing something wrong because I don't know why it commits a merge when I do a pull request.

**Comment by Ryman** [comment]
@AlexanderElias No worries, `git rebase` crash-course:

- Be on your working branch
- `git rebase -i master`, This will start an interactive rebase
- You'll see a load of commits in reverse order, with the word `pick` beside it
- if you're in `vim` then press `i` to enter insert mode
- Replace the word `pick` with `s` on all but the top line (make sure to keep the spaces!)
- exit the editor (if it's vim, press `esc`, then `:`, `w`, `enter`, `:`, `q`, `enter`)
- It'll ask you to edit the commit message for the squash, you can just delete everything and write it from scratch if you want

For a better, more thorough example, see [here](https://help.github.com/articles/using-git-rebase/)

cc @cburgdorf Perhaps you have a better link for rebase tutorials?

**Comment by AlexanderElias** [comment]
Hey @Ryman I think I made those changes as described but I still see those old commits in the commits tab. Did those get amended?

**Comment by Ryman** [comment]
@AlexanderElias I forgot to mention above, but did you push the updated branch?

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were style issues with your Pull Request:

 * Commit: 3649d7e35becaaa05863b4e7b7473a05db32217b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: e21075b1f15387a9f1d85b5ac37bc7af4771dfe4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: c41c70d98a33813b0bcd257ffeda379ea2a3c17a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 100 characters

 * Commit: 6822dc344683510d7f4def3ec78b8ff659026848
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: adc2191dd629adf6cbd2db54d5a3435c7e6a8d75
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 100 characters

 * Commit: 5b705309b5d967e5bc3ffc954fb373b44b25f6a1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/nickel-org/nickel.rs/blob/master/contributing.md

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were style issues with your Pull Request:

 * Commit: 3649d7e35becaaa05863b4e7b7473a05db32217b
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: e21075b1f15387a9f1d85b5ac37bc7af4771dfe4
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: c41c70d98a33813b0bcd257ffeda379ea2a3c17a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 100 characters

 * Commit: 6822dc344683510d7f4def3ec78b8ff659026848
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: adc2191dd629adf6cbd2db54d5a3435c7e6a8d75
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**
  * Your subject line is longer than 100 characters

 * Commit: 5b705309b5d967e5bc3ffc954fb373b44b25f6a1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 2c8691b067e83e4fa9655ecc09966258fc795c52
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/nickel-org/nickel.rs/blob/master/contributing.md

---
This message was auto-generated by https://gitcop.com

**Comment by AlexanderElias** [comment]
Yeah I was pretty sure I pushed it before. But maybe not I just did it again and the commits are showing up as new even though I squashed the others. Sorry for being such a newbie. I've used git but never collaborate yet.

**Comment by Ryman** [comment]
@AlexanderElias do you want to PM me on https://gitter.im/nickel-org/nickel.rs and we can figure this out?

**Comment by Ryman** [comment]
@homu r+

**Pull request merged**